definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js

```diff
@@ -247,6 +247,17 @@ declare module "antd" {
     validateFieldsAndScroll(options?: Object, callback?: ValidateCallback): void
   };
 
+  declare interface RcBaseFormProps {
+    wrappedComponentRef?: any;
+  }
+
+  declare interface FormComponentProps extends RcBaseFormProps {
+    form: WrappedFormUtils;
+  }
+
+  declare type FormWrappedProps<T> = <C: React$ComponentType<T>>
+  (component: C) => C;
```

Comment:
@pascalduez I think we have an issue here, the returned type is not `C` but it's more complex than that as [defined here](https://github.com/ant-design/ant-design/blob/2d8396bad79a086ae57b3ef462e681e81786f02d/components/form/interface.ts)

```javascript
// Copy from @types/react-redux https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-redux/index.d.ts
export type Matching<InjectedProps, DecorationTargetProps> = {
  [P in keyof DecorationTargetProps]: P extends keyof InjectedProps
    ? InjectedProps[P] extends DecorationTargetProps[P]
      ? DecorationTargetProps[P]
      : InjectedProps[P]
    : DecorationTargetProps[P]
};

export type GetProps<C> = C extends React.ComponentType<infer P> ? P : never;

export type ConnectedComponentClass<C extends React.ComponentType<any>, P> = React.ComponentClass<
  JSX.LibraryManagedAttributes<C, P>
> &
  hoistNonReactStatics.NonReactStatics<C> & {
    WrappedComponent: C;
  };

export type FormWrappedProps<TOwnProps extends WrappedFormInternalProps> = <
  C extends React.ComponentType<Matching<TOwnProps, GetProps<C>>>
>(
  component: C,
) => ConnectedComponentClass<C, Omit<TOwnProps, keyof WrappedFormInternalProps>>;
```

I don't know how I can fix that, any help is welcome